package.json

```diff
@@ -119,17 +117,21 @@
     "file-loader": "5.1.0",
     "html-webpack-plugin": "^5.5.0",
     "jest": "^27.3.1",
+    "jest-dom": "^4.0.0",
     "prettier": "^2.4.1",
     "pretty": "^2.0.0",
     "process": "^0.11.10",
     "raw-loader": "^4.0.0",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
+    "react": "^18.1.0",
+    "react-dom": "^18.1.0",
     "rimraf": "^3.0.0",
     "ts-jest": "^27.0.7",
     "typescript": "^4.5.2",
     "webpack": "^5.68.0",
     "webpack-cli": "^4.9.2",
     "webpack-dev-server": "^4.7.4"
+  },
+  "resolutions": {
+    "@types/react": "^18.0.12"
```

Comment:
Conflicting types in styled-system were fighting me, so I cheated and used a resolution – but let me know if you have any other ideas. Can remove if styled-system eventually updates their types dependency to something compatible with ours.

Reply:
I've had to do this before.